File: PYTHON/03-control-flow/main.py
# if / elif / else
score = 85

if score >= 90:
    print("A 학점")
elif score >= 80:
    print("B 학점")
elif score >= 70:
    print("C 학점")
else:
    print("D 학점")

# 조건부 표현식 (삼항 연산자)
age = 20
status = "성인" if age >= 19 else "미성년자"
print(status)

# for 반복문
print("--- for with range ---")
for i in range(5):         # 0, 1, 2, 3, 4
    print(i, end=" ")
print()

for i in range(2, 8, 2):   # 2, 4, 6
    print(i, end=" ")
print()

# for with list
fruits = ["사과", "바나나", "체리"]
for fruit in fruits:
    print(f"{fruit} 맛있어요!")

# enumerate: 인덱스와 값 함께 가져오기
for idx, fruit in enumerate(fruits):
    print(f"{idx}번째: {fruit}")

# while 반복문
print("--- while ---")
count = 0
while count < 3:
    print(f"count: {count}")
    count += 1

# break
print("--- break ---")
for i in range(10):
    if i == 5:
        break
    print(i, end=" ")
print()

# continue
print("--- continue ---")
for i in range(10):
    if i % 2 == 0:
        continue
    print(i, end=" ")  # 1 3 5 7 9
print()

# pass: 아무것도 하지 않음
for i in range(5):
    if i == 2:
        pass  # 나중에 구현 예정
    else:
        print(i, end=" ")
print()

File: C/03-control-flow/main.c
#include <stdio.h>

int main() {
    int score = 85;

    // if-else if-else
    if (score >= 90) {
        printf("A\n");
    } else if (score >= 80) {
        printf("B\n");
    } else if (score >= 70) {
        printf("C\n");
    } else {
        printf("F\n");
    }

    // switch
    int day = 3;
    switch (day) {
        case 1: printf("월\n"); break;
        case 2: printf("화\n"); break;
        case 3: printf("수\n"); break;
        case 4: printf("목\n"); break;
        case 5: printf("금\n"); break;
        default: printf("주말\n");
    }

    // 삼항 연산자
    int age = 20;
    char* status = (age >= 18) ? "성인" : "미성년자";
    printf("%s\n", status);

    // for문
    printf("for: ");
    for (int i = 1; i <= 5; i++) {
        printf("%d ", i);
    }
    printf("\n");

    // while문
    printf("while: ");
    int j = 1;
    while (j <= 5) {
        printf("%d ", j);
        j++;
    }
    printf("\n");

    // do-while문
    printf("do-while: ");
    int k = 1;
    do {
        printf("%d ", k);
        k++;
    } while (k <= 5);
    printf("\n");

    // 중첩 반복문 (구구단)
    printf("\n구구단:\n");
    for (int i = 2; i <= 9; i++) {
        for (int j = 1; j <= 9; j++) {
            printf("%d x %d = %d\n", i, j, i * j);
        }
        printf("\n");
    }

    // break / continue
    printf("break/continue: ");
    for (int i = 1; i <= 10; i++) {
        if (i % 2 == 0) continue;   // 짝수 건너뛰기
        if (i > 7) break;           // 7 초과면 종료
        printf("%d ", i);           // 1 3 5 7
    }
    printf("\n");

    // goto (비권장, 하지만 가끔 유용)
    int error = 0;
    if (error) {
        goto cleanup;
    }
    printf("정상 실행\n");

cleanup:
    printf("정리 작업\n");

    return 0;
}
